### PR TITLE
update groups and circular dependencies approach

### PR DESCRIPTION
**Pull request by jmhauck**

### CODE REVIEW

File: packages/common/src/interfaces.ts [hunk 1]
```diff
@@ -258,9 +272,10 @@ export interface IItemTemplateConversions {
     templateDictionary: any,
     destinationAuthentication: UserSession,
     progressTickCallback: () => void
-  ): Promise<string>;
-  postProcessCircularDependencies?(
-    newItemTemplate: IItemTemplate,
+  ): Promise<ICreateItemFromTemplateResponse>;
+  postProcessDependencies?(
+    newItemTemplate: IItemTemplate[],
```
Comment (MikeTschudi) [inline packages/common/src/interfaces.ts]: Let's please rename the `newItemTemplate` array to `templates` to match function signature, and `authentication` is clearer than the function's `destinationAuthentication`

File: packages/common/test/restHelpers.test.ts [hunk 2]
```diff
@@ -1837,9 +1837,21 @@ describe("Module `restHelpers`: common REST utility functions shared across pack
...
-    xit("shareItem", done => {
-      console.warn("========== TODO ==========");
-      done.fail();
+    it("can handle error on shareItem", done => {
+      const groupId: string = "grp1234567890";
+      const id: string = "itm1234567890";
+      fetchMock
+        .get(
+          "https://myorg.maps.arcgis.com/sharing/rest/community/users/casey?f=json&token=fake-token",
+          MOCK_USER_SESSION.token
```
Comment (MikeTschudi) [inline packages/common/test/restHelpers.test.ts]: Why `MOCK_USER_SESSION.token` as the response, please? Doesn't match what AGO would return for a user query like this.

File: packages/common/src/interfaces.ts (hunk 1 above)
Comment (jmhauck) [reply packages/common/src/interfaces.ts]: good catch..will do

File: packages/common/test/restHelpers.test.ts (hunk 2 above)
Comment (jmhauck) [reply packages/common/test/restHelpers.test.ts]: oops...not intended

**Pull request merged**